### PR TITLE
Bump Pyenv's Python to 3.9

### PR DESCRIPTION
**Pull request by Sjors**
Since the release notes mention support for Python 3.9. Or do we also want to guarantee backwards compatibility?

Using 3.9.1 for PyEnv (latest available on macOS) and 3.9.2 for the Windows installer.

### CODE REVIEW

**Comment by Sjors** [comment]
Seeing some timeouts.

**Comment by achow101** [comment]
We want to guarantee backwards compatibility so everything should still use the oldest python version supported. The release note is just mentioning that people using Python 3.9 in their projects will be able to use HWI.

**Comment by Sjors** [comment]
Ok, in that case I think it's sufficient to bump `.python-version` for PyEnv users.

**Comment by Sjors** [comment]
Some confusing CI failings.

**Comment by achow101** [comment]
Now that we have CI for all current python minor versions, we can bump pyenv and the build python versions to something more recent. Can you please rebase this and bump to more recent python?

**Comment by Sjors** [comment]
Done, let's see how 3.9.7 fares (3.9.8 isn't available on Homebrew yet, so Cirrus CI probably also won't have it)

**Comment by Sjors** [comment]
Done, let's see how 3.9.7 fares (3.9.8 isn't available on Homebrew yet, so Cirrus CI probably also won't have it)

**Comment by Sjors** [comment]
The three test failures seem to be due to a timeout.

**Comment by achow101** [comment]
ACK 3117eb4ffde8b5fad0f8da0ba4c153e492ce9612

File: .cirrus.yml [hunk 1]
```diff
@@ -214,6 +214,7 @@ dist_builder_task:
   container:
     dockerfile: contrib/build.Dockerfile
   build_script:
+    - pyenv install 3.9.7
```
Comment (achow101) [inline .cirrus.yml:217]: `contrib/build.Dockerfile` should be changed instead of cirrus.

**Comment by achow101** [comment]
`contrib/build_wine.sh` should also be updated to use 3.9.7

File: .cirrus.yml (hunk 1 above)
Comment (Sjors) [reply .cirrus.yml:217]: Should this line be dropped entirely then?

Comment (achow101) [reply .cirrus.yml:217]: Yes

Comment (Sjors) [reply .cirrus.yml]: Done

**Comment by achow101** [comment]
ACK d1055df7ba60ff6ba2a182980dcab1799c239c7a

**Pull request merged**